src/Abc.Zebus.Persistence.RocksDb/RocksDbStorage.cs

```diff
@@ -0,0 +1,242 @@
+        {
+            var nonAcked = entry.Aggregate(0, (s, e) => s + (e.IsAck ? -1 : 1));
+            var peerKey = GetPeerKey(entry.Key);
+            using (var iterator = _db.NewIterator(_peersColumnFamily))//, new ReadOptions().SetTotalOrderSeek(true)))
+            {
+                // TODO: figure out why Seek() returns true for a different key
+                var alreadyExists = iterator.Seek(peerKey).Valid() && CompareStart(iterator.Key(), peerKey, peerKey.Length);
+                var currentNonAcked = alreadyExists ? BitConverter.ToInt32(iterator.Value(), 0) : 0;
+
+                var value = currentNonAcked + nonAcked;
+                _db.Put(peerKey, BitConverter.GetBytes(value), _peersColumnFamily);
+            }
+        }
+
+        public IMessageReader CreateMessageReader(PeerId peerId)
+        {
+            using (var iterator = _db.NewIterator(_peersColumnFamily))
+            {
+                if (!iterator.Seek(GetPeerKey(peerId)).Valid())
+                    return null;
+            }
+
+            return new RocksDbMessageReader(_db, peerId, _messagesColumnFamily);
+        }
+
+        public Task RemovePeer(PeerId peerId)
+        {
+            var key = CreateKeyBuffer(peerId);
+            FillKey(key, peerId, 0, Guid.Empty);
+
+            using (var cursor = _db.NewIterator(_messagesColumnFamily))
+            {
+                if (!cursor.Seek(key).Valid())
+                    return Task.CompletedTask;
+
+                var peerIdLength = peerId.ToString().Length;
+                byte[] currentKey;
+                do
+                {
+                    currentKey = cursor.Key();
+                    _db.Remove(currentKey);
+                    cursor.Next();
+                } while (cursor.Valid() && CompareStart(currentKey, key, peerIdLength));
+            }
+
+            _db.Remove(GetPeerKey(peerId), _peersColumnFamily);
+
+            // TODO: remove out of order acks
+            return Task.CompletedTask;
+        }
+
+        public Dictionary<PeerId, int> GetNonAckedMessageCounts()
+        {
+            var nonAckedCounts = new Dictionary<PeerId, int>();
+            using (var cursor = _db.NewIterator(_peersColumnFamily))
+            {
+                for (cursor.SeekToFirst(); cursor.Valid(); cursor.Next())
+                {
+                    var peerId = ReadPeerKey(cursor.Key());
+                    nonAckedCounts[peerId] = BitConverter.ToInt32(cursor.Value(), 0);
+                }
+            }
+
+            return nonAckedCounts;
+        }
+
+        public static void FillKey(byte[] key, PeerId peerId, long ticks, Guid messageId)
+        {
+            var peerPart = Encoding.UTF8.GetBytes(peerId.ToString());
+            Buffer.BlockCopy(peerPart, 0, key, 0, peerPart.Length);
+
+            var tickPart = BitConverter.GetBytes(ticks);
+            Array.Reverse(tickPart); // change endianness
```

Comment:
I suppose `BitConverter.IsLittleEndian` could be used if we want to run this code on ARM :)